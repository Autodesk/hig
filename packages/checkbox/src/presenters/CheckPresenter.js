/* eslint-disable react/no-danger */
import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import icons from "@hig/icons";

import "./CheckPresenter.scss";

export default class CheckPresenter extends Component {
  static propTypes = {
    /**
     * Checks the checkbox
     */
    checked: PropTypes.bool,
    /**
     * Prevents user actions on the checkbox
     */
    disabled: PropTypes.bool,
    /**
     * Sets indeterminate state for checkbox
     */
    indeterminate: PropTypes.bool
  };

  static defaultProps = {
    checked: false,
    disabled: false,
    indeterminate: false
  };

  render() {
    const { checked, disabled, indeterminate } = this.props;

    const iconClasses = cx([
      "hig__checkbox__wrapper",
      {
        "hig__checkbox__wrapper--checked": checked,
        "hig__checkbox__wrapper--indeterminate": indeterminate,
        "hig__checkbox__wrapper--disabled": disabled
      }
    ]);

    return (
      <span className={iconClasses}>
        <span
          className="hig__checkbox__wrapper__check"
          dangerouslySetInnerHTML={{ __html: icons["check-white-24"] }}
        />
        <span
          className="hig__checkbox__wrapper__check--indeterminate"
          dangerouslySetInnerHTML={{
            __html: icons["checkmark-indeterminate-16"]
          }}
        />
        <span
          className="hig__checkbox__wrapper__check--disabled"
          dangerouslySetInnerHTML={{ __html: icons["check-disabled-16"] }}
        />
      </span>
    );
  }
}
