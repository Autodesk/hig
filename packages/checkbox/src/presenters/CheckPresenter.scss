@import '~@hig/styles/mixins/colors';

.hig__checkbox__wrapper__check--indeterminate,
.hig__checkbox__wrapper__check--disabled,
.hig__checkbox__wrapper__check {
  display: none;
}

.hig__checkbox__wrapper--checked {
  .hig__checkbox__wrapper__check {
    display: block;
  }
}

.hig__checkbox__wrapper--checked.hig__checkbox__wrapper--disabled {
  .hig__checkbox__wrapper__check {
    display: none;
  }

  .hig__checkbox__wrapper__check--disabled {
    display: block;
  }
}

.hig__checkbox__wrapper--indeterminate,
.hig__checkbox__wrapper--indeterminate.hig__checkbox__wrapper--checked {
  .hig__checkbox__wrapper__check--indeterminate {
    display: block;
    background-color: color(hig-slate-40);
  }

  .hig__checkbox__wrapper__check {
    display: none;
  }

}

.hig__checkbox__inputV1:focus + .hig__checkbox__wrapper {
  .hig__checkbox__wrapper__check--indeterminate {
    background-color: color(hig-blue-60);
  }
}


// workaround for image sizes

.hig__checkbox__wrapper--checked .hig__checkbox__wrapper__check {
  svg {
    top: -2px;
    position: relative;
  }
}

.hig__checkbox__wrapper--checked .hig__checkbox__wrapper__check--disabled {
  svg {
    top: -3px;
    position: relative;
  }
}

.hig__checkbox__wrapper__check--indeterminate {
  width: 16px;
  height: 16px;
  svg {
    top: -6px;
    position: relative;
  }
}

// end workaround