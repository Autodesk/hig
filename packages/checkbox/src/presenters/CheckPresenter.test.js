import React from "react";
import renderer from "react-test-renderer";
import CheckPresenter from "./CheckPresenter";

describe("checkbox/presenters/CheckPresenter", () => {
  [
    {
      description: "renders without props",
      props: {}
    },
    {
      description: "renders with all props",
      props: {
        checked: true,
        disabled: true,
        indeterminate: false
      }
    }
  ].forEach(({ description, props: { children, ...otherProps } }) => {
    it(description, () => {
      const presenter = (
        <CheckPresenter {...otherProps}>{children}</CheckPresenter>
      );
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
