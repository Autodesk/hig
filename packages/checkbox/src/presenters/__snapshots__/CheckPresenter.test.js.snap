// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox/presenters/CheckPresenter renders with all props 1`] = `
<span
  className="hig__checkbox__wrapper hig__checkbox__wrapper--checked hig__checkbox__wrapper--disabled"
>
  <span
    className="hig__checkbox__wrapper__check"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width=\\"11\\" height=\\"9\\" viewBox=\\"0 0 11 9\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"check-white-24\\"><polygon points=\\"7.13516024 12 4 8.9767 4.97189967 7.9722 7.11774268 10.0281 12.0092894 5 13 5.9863\\" transform=\\"translate(-3 -4)\\" stroke-width=\\".5\\" stroke=\\"#FFF\\" fill=\\"#FFF\\" fill-rule=\\"evenodd\\"/></svg>",
      }
    }
  />
  <span
    className="hig__checkbox__wrapper__check--indeterminate"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width=\\"8\\" height=\\"2\\" viewBox=\\"0 0 8 2\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"checkmark-indeterminate-16\\"><rect x=\\"4\\" y=\\"7\\" width=\\"8\\" height=\\"2\\" transform=\\"translate(-4 -7)\\" fill=\\"#FFF\\" fill-rule=\\"evenodd\\"/></svg>",
      }
    }
  />
  <span
    className="hig__checkbox__wrapper__check--disabled"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width=\\"10\\" height=\\"8\\" viewBox=\\"0 0 10 8\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"check-disabled-16\\"><polygon points=\\"2.95073905 6.58823529 2.274e-13 3.74277647 0.914729106 2.79736471 2.93434606 4.73232941 7.5381547 2.274e-13 8.47058824 0.928282353\\" transform=\\"translate(1 1)\\" stroke-width=\\".5\\" stroke=\\"#D1DDEE\\" fill=\\"#D1DDEE\\" fill-rule=\\"evenodd\\"/></svg>",
      }
    }
  />
</span>
`;

exports[`checkbox/presenters/CheckPresenter renders without props 1`] = `
<span
  className="hig__checkbox__wrapper"
>
  <span
    className="hig__checkbox__wrapper__check"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width=\\"11\\" height=\\"9\\" viewBox=\\"0 0 11 9\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"check-white-24\\"><polygon points=\\"7.13516024 12 4 8.9767 4.97189967 7.9722 7.11774268 10.0281 12.0092894 5 13 5.9863\\" transform=\\"translate(-3 -4)\\" stroke-width=\\".5\\" stroke=\\"#FFF\\" fill=\\"#FFF\\" fill-rule=\\"evenodd\\"/></svg>",
      }
    }
  />
  <span
    className="hig__checkbox__wrapper__check--indeterminate"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width=\\"8\\" height=\\"2\\" viewBox=\\"0 0 8 2\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"checkmark-indeterminate-16\\"><rect x=\\"4\\" y=\\"7\\" width=\\"8\\" height=\\"2\\" transform=\\"translate(-4 -7)\\" fill=\\"#FFF\\" fill-rule=\\"evenodd\\"/></svg>",
      }
    }
  />
  <span
    className="hig__checkbox__wrapper__check--disabled"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width=\\"10\\" height=\\"8\\" viewBox=\\"0 0 10 8\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"check-disabled-16\\"><polygon points=\\"2.95073905 6.58823529 2.274e-13 3.74277647 0.914729106 2.79736471 2.93434606 4.73232941 7.5381547 2.274e-13 8.47058824 0.928282353\\" transform=\\"translate(1 1)\\" stroke-width=\\".5\\" stroke=\\"#D1DDEE\\" fill=\\"#D1DDEE\\" fill-rule=\\"evenodd\\"/></svg>",
      }
    }
  />
</span>
`;
